shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -118,6 +118,10 @@ - (instancetype)initWithEngine:(FlutterEngine*)engine
                        nibName:(nullable NSString*)nibName
                         bundle:(nullable NSBundle*)nibBundle {
   NSAssert(engine != nil, @"Engine is required");
+  if (!engine.shell.IsSetup()) {
```

Comment:
Thanks for the feedback @ColdPaleLight .

I've changed the code so now its checking that the shell is setup from the FlutterEngine and logging an error if this is not the case.

Is this correct?

Reply:
Also if this code addresses the issue I will update the unit tests as they are currently failing for the FlutterViewController

Reply:
I guess we can add a flag to the `FlutterEngine` to determine whether the `FlutterEngine` is running.  If one of these two methods is executed, the flag will be set to `YES`, indicating that the engine has been running.

https://github.com/flutter/engine/blob/70798c3174c9bcfc6d63deb9e6e9a105ebd0949f/shell/platform/darwin/ios/framework/Source/FlutterEngine.mm#L572

https://github.com/flutter/engine/blob/70798c3174c9bcfc6d63deb9e6e9a105ebd0949f/shell/platform/darwin/ios/framework/Source/FlutterEngine.mm#L1087-L1090

You might want to try @gaaclarke for better ideas.

Reply:
I believe Shell::IsSetup is an appropriate proxy for asking if a FlutterEngine has been run.  What I don't like in this PR is that the check here is very distant from the actual requirement that the engine is run.  I'd rather we had the plumbing in place to handle the error at the location of the requirement.  At the very least we should have a comment here that explains why it is a requirement since the code around it doesn't explain it.

Also, the error message should be a sentence and use `[engine run]` as an example since there are a few run methods, not just that one.   I'd say something like "has not been run (example: -[FlutterEngine run])".

Reply:
> I believe Shell::IsSetup is an appropriate proxy for asking if a FlutterEngine has been run.

I'm concerned that this condition is not precise enough, there may be cases where FlutterEngine's `Shell::IsSetup` is  true, but the engine is not running yet. for example
```
FlutterViewController *flutterVC1 = [[FlutterViewController alloc] initWithProject:nil nibName:nil bundle:nil];
FlutterEngine *engine = flutterVC1.engine;
engine.viewController = nil;
```
In this case, this line of code is called https://github.com/flutter/engine/blob/d91489e9f62fcdab9d43783863c8c46b1e33b236/shell/platform/darwin/ios/framework/Source/FlutterViewController.mm#L210
But this line of code is not called
https://github.com/flutter/engine/blob/d91489e9f62fcdab9d43783863c8c46b1e33b236/shell/platform/darwin/ios/framework/Source/FlutterViewController.mm#L643
So we will get an `FlutterEngine` with a shell set up but not yet running.Please let me know if I missed something.

Reply:
Yep, I missed that.  If we plumb the check to where it is actually required, we'll be able to address that.  It just reinforces that need to me.

Reply:
@gaaclarke did you mean to add the check after viewDidAppear is called?
https://github.com/flutter/engine/blob/64e78f33102610cd7bc9dae4be04433e33073bc2/shell/platform/darwin/ios/framework/Source/FlutterViewController.mm#L713
So check here if the shell IsSetup (and if not log that error) before executing https://github.com/flutter/engine/blob/64e78f33102610cd7bc9dae4be04433e33073bc2/shell/platform/darwin/ios/framework/Source/FlutterViewController.mm#L715

Reply:
I didn't note what assert is failing in the issue.  I can't remember where it is.  Wherever that assert is happening, we'll want to get the check as close as possible.  At some point the code starts executing down a doomed path.  We want to put the error message as close to the assert as possible but at a location where we have enough context to understand we are about to execute a doomed path.

The location of the check seems a bit high in the call stack (the assert seems too low).  Maybe it would help if we noted the stacktrace of the assert so we can better reason about where it makes the most sense.